update Sample-Level column name (#4)
ismms-himc/cellranger-aws-pipeline-rebuild: 10x_Experimental_Scenarios.md

# 10x Experimental Scenarios

> This document discusses proposed changes to the various spreadsheets used by HIMC techs who are running the assay, the computational team performing pre-processing, and the Cell Ranger software (e.g. we should shift to the most general input spreadsheet as arguments). This document heavily references the [10x documentation](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/using/mkfastq), defines schemas for the various spreadsheets used by all parties, and contains several example spreadsheets that represent different common experimental scenarios.

Reviewer: Would suggest keeping the opening as punchy as possible; here's an attempt:
```suggestion
This document proposes schemas and vocabulary to be used across: 
* spreadsheets used by HIMC techs running assays
* the computational team's pre-processing pipelines
* the Cell Ranger software

Below are schemas and example data for various spreadsheets used by all parties, and [a glossary](#glossary) of relevant terms. This document heavily references the [10x documentation](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/using/mkfastq) as well.
```

> # Glossary

Reviewer: I would vote for putting the Glossary at the end, but with a good call-out at the beginning.

> The relationships between components in 10x single cell assay can be complicated. As a first step, we define a common vocabulary for ourselves below. We have ordered entities based on their natural progression during the experiment and data-processing steps.  

Reviewer: ```suggestion
The relationships between components in 10x's single cell assay can be complicated. As a first step, we define a common vocabulary below, in rough order of appearance during the experiment and data-processing steps.
```

> * **biological samples**: a cell suspension extracted from a single biological source, see [10x-Glossary](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/glossary)

Reviewer: ```suggestion
* **biological sample**: a cell suspension extracted from a single biological source, see [10x-Glossary](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/glossary)
```

> * **loading sample**: the sample loaded into a 10x lane, which is usually the biological sample but may be a hash of many biological samples

Reviewer: ```suggestion
* **loading sample**: a sample loaded into a 10x lane (usually 1:1 with biological sample, but may be a hash of many biological samples)
```

> * **sequencing libraries**: the end product(s) from a 10x chip lane. A single 10x lane can produce a single (e.g. GEX) or multiple sequencing libraries (e.g. GEX and VDJ; or GEX, ADT, and HTO). These sequencing libraries will either be pooled into a single pooled library and sequenced or run on separate sequencing runs (e.g. as is the case for VDJ and GEX which require different sequencing conditions)

Reviewer: ```suggestion
* **sequencing libraries**: end product(s) from a 10x chip lane. A single 10x lane can produce a single (e.g. GEX) or multiple sequencing libraries (e.g. {GEX, VDJ} or {GEX, ADT, HTO}). These sequencing libraries will either be pooled into a single pooled library and sequenced or sequenced in separate sequencing runs (as in the case of {GEX, VDJ}, where each library requires different sequencing conditions)
```

* **pooled library**: a combination of several indexed sequencing libraries (from several 10x chip lanes) for loading into a sequencer and obtaining a BCL file

> * **BCL files**: contains sequencing information on a pooled library (e.g. set of multiplexed sequencing libraries)

Reviewer: ```suggestion
* **BCL file**: contains sequencing information on a pooled library (e.g. set of multiplexed sequencing libraries)
```

* **FASTQs**: the products of de-multiplexing BCL files

* **Seq-Run-FASTQ Set**: a set of FASTQs that have been de-multiplexed from a single BCL file that give data for a single indexed sequencing library (e.g. a single GEX or ADT library). This set consists of lane- and read-specific FASTQs. 

> * **FBM-FASTQ Set**: a set of FASTQs that will be used to generate a single FBM (feature barcode matrix). If the same pooled library is sequenced multiple times (producing multiple BCL files) we will need to combine multiple Seq-Run-FASTQ sets (see above) into a single FBM.

Reviewer: ```suggestion
* **FBM-FASTQ Set**: a set of FASTQs that will be used to generate a single FBM (feature barcode matrix). If the same pooled library is sequenced multiple times (producing multiple BCL files) we will need to combine multiple **Seq-Run-FASTQ Sets** (see above) into a single FBM.
```

> * **Processing-Run**: a set of cellranger mkfastq and count runs that take as input: 1) one or more BCL files and 2) Processing-Run Input CSV files. The Processing-Run produces the following outputs: 1) FASTQs, 2) FBMs (feature barcode matrices) and if applicable TCR/BCR data, 3) [Processing-Run Status CSV] which lists out all jobs in the Processing-Run as well as where to find outputs 4) [Processing-Run Meta-Data CSV] relevant metadata (still being sorted out) for 10x techs.

Reviewer: ```suggestion
* **Processing-Run**: a set of `cellranger` `mkfastq` and `count` runs that take as input:
1) one or more BCL files
2) Processing-Run Input CSV files.

and produces outputs:
1) FASTQs
2) FBMs (feature barcode matrices) and, if applicable, TCR/BCR data
3) [Processing-Run Status CSV] which lists out all jobs in the Processing-Run (as well as where to find outputs) 
4) [Processing-Run Meta-Data CSV] relevant metadata (still being sorted out) for 10x techs).
```

> * **Cell Ranger outputs**: Feature barcode matrix (FBM) which can be the product of several sequencing runs and BCL files, TCR/BCR contigs

Reviewer: Is 1 **Cell Ranger output** always equal 1 **FBM**? If so, I'd just say FBM everywhere…

Also, nit:
```suggestion
* **Cell Ranger output**: Feature barcode matrix (FBM) which can be the product of several sequencing runs and BCL files, TCR/BCR contigs
```

> * **Sample-Level Outputs**: FBMs and VDJ data that has been re-organized into sample level data. This usually consists of copying sample level data out of the Processing Run, but for hashing runs requires manual de-hashing (post-Cell Ranger step).

Reviewer: ```suggestion
* **Sample-Level Outputs**: FBMs and VDJ data that has been re-organized into sample level data. Usually generated by copying sample-level data out of the Processing Run, but for hashing runs requires manual de-hashing (a post-Cell Ranger step).
```

is the manual de-hashing documented? would be good to link to. same with "Processing Run" (intra-doc link would be ideal, here and elsewhere)

# 10x Examples from Documentation

> Below are two experimental scenarios from 10x that have been paraphrased into our vocabulary (see [Glossary]). 

Reviewer: ```suggestion
Below are two experimental scenarios from 10x that have been paraphrased in our vocabulary (see [Glossary]). 
```

> In this example, we have two 10x libraries (each processed through a separate Chromium chip channel) that are multiplexed on a single flowcell. Note that after running `cellranger mkfastq`, we run a separate instance of the pipeline on each library

### 1 Sample, 1 Seq-Library, 2 Seq-Run, 1 FBM
[![Schematic of 1-sample, 2-seq-run workflow](https://support.10xgenomics.com/img/mkfastq-2.png "")](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/using/mkfastq#example_workflows)

> In this example, we have one 10x library sequenced on two flowcells. Note that after running cellranger mkfastq, we run a single instance of the pipeline on all the FASTQ files generated.

The common scenario of sequencing the same 'sequencing library' more than once (in the above example) is the reason why the 10x technicians are moving towards a FASTQ-level organization (rather than a 'sequencing library' a.k.a. 'pooled library' level).

# 10X Technician Spreadsheets

> Below are 4 proposed spreadsheets for use by the 10X techs (not all experiment-related columns are shown). The spreadsheets produced in these three sections ([10x Technician Spreadsheets], [Processing-Run CSVs], and [Cell Ranger Required CSVs]) are all based on the same example: Four biological samples that are hashed together, measure three surface markers (3 ADTs), and are run on a single 10x chip lane. 

Reviewer: ```suggestion
Below are 4 proposed spreadsheets for use by the 10X techs (not all experiment-related columns are shown):
- [Sample-Level Spreadsheet]
- [10x Technician Spreadsheets]
- [Processing-Run CSVs]
- [Cell Ranger Required CSVs]

The examples in the latter three sections are all based on the same hypothetical analysis: four biological samples that are hashed together, with three surface markers measured (3 ADTs), and run on a single 10x chip lane. 
```

## 1. Sample-Level Spreadsheet
| Sample Name | Loading Sample | Expected Cell Number | Reference Transcriptome | Chemistry | HTO | Library Features |
|---|---|---|---|---|---|---|
| S1  | H1 | 3000 | GRCh38 | 3p | HTO-1 | LF-1 |
| S2  | H1 | 4000 | GRCh38 | 3p | HTO-2 | LF-1 |
| S3  | H1 | 5000 | GRCh38 | 3p | HTO-3 | LF-1 |
| S4  | H1 | 6000 | GRCh38 | 3p | HTO-4 | LF-1 |

### Columns
- `Sample Name`: the name of the biological sample being processed (see [Glossary])
- `Loading Sample`: the name of the sample being loaded into the 10x chip (see [Glossary])
- `Expected Cell Number`: the estimated number of cells in the loaded sample
- `Reference Transcriptome`: the reference transcriptome that reads are aligned to (very big files on AWS S3 buckets)
- `Chemistry`: the name of the 10x kit chemistry being used (e.g. 5-prime)
- `HTO`: the name of the hash tag oligo (HTO) that is used to label this sample, the value will be `-` for a non-hashed sample
- `Library Features`: this links a sample to its list of features in the [Library Features Table]. The value is `-` if we are not measuring any ADTs or HTOs

> ### Explanation of this spreadsheet

Reviewer: I would shorten this and make it "Discussion":
```suggestion
### Discussion
```

This spreadsheet shows the list of `Library Features` that are associated with a loading sample (e.g. `H1`) and its subsequent sequencing libraries (`GEX`, `ADT`, `HTO`). This list is linked to specific samples and libraries in the two spreadsheets: [Sample-Level Spreadsheet] and [FASTQ-Level Spreadsheet].

## 4. Features Table
| HIMC Feature Name | Chemistry | Oligo ID | Oligo Sequence |  
|---|---|---|---|
| HTO-1_H-101_3p_Lot-# | 3p | H-101 | ACTG |  
| HTO-2_H-102_3p_Lot-# | 3p | H-102 | ACTG |  
| HTO-3_H-103_3p_Lot-# | 3p | H-103 | ACTG |  
| HTO-4_H-104_3p_Lot-# | 3p | H-104 | ACTG |  
|   CD3_A-101_3p_Lot-# | 3p | A-101 | ACTG |  
|   CD4_A-102_3p_Lot-# | 3p | A-102 | ACTG |  
|   CD5_A-103_3p_Lot-# | 3p | A-103 | ACTG |  
| ... | ... | ... | ... |  

### Columns
- `HIMC Feature Name`: name of a feature (ADT or HTO) being measured
- `Chemistry`: the type of chemistry this feature is made for
- `Oligo`: the human readable name of the oligo used to label this feature
- `Sequence`: the actual oligo sequence 

### Explanation of this spreadsheet
This spreadsheet contains all features being used by the HIMC and each `HIMC Feature Name` is unique. 

# Processing-Run CSVs

> A `Processing-Run` takes as input two spreadsheets (produced by the 10x techs using value-lookups) and one or more BCLs. The two spreadsheets are similar to the required spreadsheets that `cellranger mkfastq` and `count` take as inputs, but also contain additional information (e.g. expected cell count) as well as an implicit layout of running all jobs required to complete a  `Processing-Run` set of jobs.

Reviewer: ```suggestion
A `Processing-Run` takes as input two spreadsheets (produced by the 10x techs using value-lookups) and one or more BCLs. The two spreadsheets are similar to the required spreadsheets that `cellranger mkfastq` and `count` take as inputs, but also contain additional information (e.g. expected cell count) as well as an implicit plan for running all jobs required to complete a  [`Processing-Run`].
```
